contracts/distribution.js

```diff
@@ -0,0 +1,287 @@
+    && api.assert(isSignedWithActiveKey === true, 'you must use a transaction signed with your active key')) {
+    const exDist = await api.db.findOne('batches', { _id: id });
+    if (api.assert(exDist, 'distribution not found') && await validateMinPayout(tokenMinPayout)
+      && await validateRecipients(tokenRecipients)) {
+      exDist.tokenMinPayout = tokenMinPayout;
+      exDist.tokenRecipients = tokenRecipients;
+      await api.db.update('batches', exDist);
+
+      // burn the token creation fees
+      if (api.sender !== api.owner && api.BigNumber(distUpdateFee).gt(0)) {
+        await api.executeSmartContract('tokens', 'transfer', {
+          // eslint-disable-next-line no-template-curly-in-string
+          to: 'null', symbol: "'${CONSTANTS.UTILITY_TOKEN_SYMBOL}$'", quantity: distUpdateFee, isSignedWithActiveKey,
+        });
+      }
+      api.emit('update', { id: exDist._id });
+    }
+  }
+};
+
+actions.setActive = async (payload) => {
+  const {
+    id,
+    active,
+    isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(isSignedWithActiveKey === true, 'you must use a custom_json signed with your active key')) {
+    return;
+  }
+  const dist = await api.db.findOne('batches', { _id: id });
+  if (api.assert(dist, 'distribution id not found')
+    && api.assert(dist.creator === api.sender || api.owner === api.sender, 'you must be the creator of this distribution')) {
+    dist.active = !!active;
+    await api.db.update('batches', dist);
+  }
+};
+
+actions.deposit = async (payload) => {
+  const {
+    id, symbol, quantity, isSignedWithActiveKey,
+  } = payload;
+
+  if (!api.assert(isSignedWithActiveKey === true, 'you must use a custom_json signed with your active key')
+    && !api.assert(quantity && api.BigNumber(quantity).dp() <= 3 && api.BigNumber(quantity).gt(0), 'invalid quantity')) {
+    return;
+  }
+
+  const dist = await api.db.findOne('batches', { _id: id });
+  if (api.assert(dist, 'distribution id not found') && api.assert(dist.active, 'distribution must be active to deposit')
+    && api.assert(validateIncomingToken(dist, symbol), `${symbol} is not accepted by this distribution`)) {
+    // deposit requested tokens to contract
+    const res = await api.executeSmartContract('tokens', 'transferToContract', { symbol, quantity, to: 'distribution' });
+    if (res.errors === undefined
+      && res.events && res.events.find(el => el.contract === 'tokens' && el.event === 'transferToContract' && el.data.from === api.sender && el.data.to === 'distribution' && el.data.quantity === quantity) !== undefined) {
+      // update token balances
+      if (dist.tokenBalances) {
+        let hasBalance = false;
+        for (let i = 0; i < dist.tokenBalances.length; i += 1) {
+          if (dist.tokenBalances[i].symbol === symbol) {
+            dist.tokenBalances[i].quantity += quantity;
+            hasBalance = true;
+            break;
+          }
+        }
```

Comment:
optional: change to use find()

Reply:
Still not sure if using find() makes it more readable or not, I do believe however its a slower operation. I have left this alone for now.

Reply:
find is definitely more readable (no manual breaking and assigning values), and for the case where so few elements are involved, it does not matter.  (I also am searching around and I don't actually think it is slower depending on the engine)